app/views/layouts/application.html.erb

```diff
@@ -10,7 +10,13 @@
   <body>
   <div class="header">
     <div class="header-container">
-      <div class="user-controls">
+      <div class="search-block">
```

Comment:
You shouldn't put any logic into layout. Layout is used to have a very basic markup that must be on every page. Like header, footer and base content wrapper. You can extract this block into a partial